From bda0979bd5d093a904d823cafac6bf0801ce4f4c Mon Sep 17 00:00:00 2001
From: Amelie Delaunay <amelie.delaunay@foss.st.com>
Date: Fri, 3 Oct 2025 17:47:59 +0200
Subject: [PATCH] ARM: dts: stm32: add phase tags marking on stm32mp15 eval
 board

bootph-all/bootph-some-ram/bootph-pre-ram as phase tags were added to
dt-schema (dtschema/schemas/bootph.yaml) to describe various node usage
during boot phases with DT.

Describe them for STM32MP15 EVAL boot devices.

Change-Id: I501dc5f5bec0934c2032648941047608cce3579a
Signed-off-by: Amelie Delaunay <amelie.delaunay@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/496298
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Patrice CHOTARD <patrice.chotard@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 arch/arm/boot/dts/st/stm32mp157a-ed1.dts | 11 ++++++++
 arch/arm/boot/dts/st/stm32mp157a-ev1.dts | 36 ++++++++++++++++++++++++
 arch/arm/boot/dts/st/stm32mp157c-ed1.dts | 11 ++++++++
 arch/arm/boot/dts/st/stm32mp157c-ev1.dts | 36 ++++++++++++++++++++++++
 arch/arm/boot/dts/st/stm32mp157d-ed1.dts | 11 ++++++++
 arch/arm/boot/dts/st/stm32mp157d-ev1.dts | 36 ++++++++++++++++++++++++
 arch/arm/boot/dts/st/stm32mp157f-ed1.dts | 11 ++++++++
 arch/arm/boot/dts/st/stm32mp157f-ev1.dts | 36 ++++++++++++++++++++++++
 8 files changed, 188 insertions(+)

diff --git a/arch/arm/boot/dts/st/stm32mp157a-ed1.dts b/arch/arm/boot/dts/st/stm32mp157a-ed1.dts
index 5d17d6c10dde..9a3e5e04d1ee 100644
--- a/arch/arm/boot/dts/st/stm32mp157a-ed1.dts
+++ b/arch/arm/boot/dts/st/stm32mp157a-ed1.dts
@@ -419,6 +419,7 @@ timer@5 {
 };
 
 &uart4 {
+	bootph-all;
 	pinctrl-names = "default", "sleep", "idle";
 	pinctrl-0 = <&uart4_pins_a>;
 	pinctrl-1 = <&uart4_sleep_pins_a>;
@@ -428,6 +429,16 @@ &uart4 {
 	status = "okay";
 };
 
+&uart4_pins_a {
+	bootph-all;
+	pins1 {
+		bootph-all;
+	};
+	pins2 {
+		bootph-all;
+	};
+};
+
 &usbotg_hs {
 	vbus-supply = <&vbus_otg>;
 };
diff --git a/arch/arm/boot/dts/st/stm32mp157a-ev1.dts b/arch/arm/boot/dts/st/stm32mp157a-ev1.dts
index 9bcf84f705a4..a390e316d027 100644
--- a/arch/arm/boot/dts/st/stm32mp157a-ev1.dts
+++ b/arch/arm/boot/dts/st/stm32mp157a-ev1.dts
@@ -564,6 +564,7 @@ &m_can1 {
 };
 
 &qspi {
+	bootph-pre-ram;
 	pinctrl-names = "default", "sleep";
 	pinctrl-0 = <&qspi_clk_pins_a
 		     &qspi_bk1_pins_a
@@ -599,6 +600,41 @@ flash1: flash@1 {
 	};
 };
 
+&qspi_clk_pins_a {
+	bootph-pre-ram;
+	pins {
+		bootph-pre-ram;
+	};
+};
+
+&qspi_bk1_pins_a {
+	bootph-pre-ram;
+	pins {
+		bootph-pre-ram;
+	};
+};
+
+&qspi_cs1_pins_a {
+	bootph-pre-ram;
+	pins {
+		bootph-pre-ram;
+	};
+};
+
+&qspi_bk2_pins_a {
+	bootph-pre-ram;
+	pins {
+		bootph-pre-ram;
+	};
+};
+
+&qspi_cs2_pins_a {
+	bootph-pre-ram;
+	pins {
+		bootph-pre-ram;
+	};
+};
+
 &sai2 {
 	clocks = <&rcc SAI2>, <&rcc PLL3_Q>, <&rcc PLL3_R>;
 	pinctrl-names = "default", "sleep";
diff --git a/arch/arm/boot/dts/st/stm32mp157c-ed1.dts b/arch/arm/boot/dts/st/stm32mp157c-ed1.dts
index 849227aa47f4..a48d9f5a2981 100644
--- a/arch/arm/boot/dts/st/stm32mp157c-ed1.dts
+++ b/arch/arm/boot/dts/st/stm32mp157c-ed1.dts
@@ -423,6 +423,7 @@ timer@5 {
 };
 
 &uart4 {
+	bootph-all;
 	pinctrl-names = "default", "sleep", "idle";
 	pinctrl-0 = <&uart4_pins_a>;
 	pinctrl-1 = <&uart4_sleep_pins_a>;
@@ -432,6 +433,16 @@ &uart4 {
 	status = "okay";
 };
 
+&uart4_pins_a {
+	bootph-all;
+	pins1 {
+		bootph-all;
+	};
+	pins2 {
+		bootph-all;
+	};
+};
+
 &usbotg_hs {
 	vbus-supply = <&vbus_otg>;
 };
diff --git a/arch/arm/boot/dts/st/stm32mp157c-ev1.dts b/arch/arm/boot/dts/st/stm32mp157c-ev1.dts
index 321346453719..92f97f505df2 100644
--- a/arch/arm/boot/dts/st/stm32mp157c-ev1.dts
+++ b/arch/arm/boot/dts/st/stm32mp157c-ev1.dts
@@ -564,6 +564,7 @@ &m_can1 {
 };
 
 &qspi {
+	bootph-pre-ram;
 	pinctrl-names = "default", "sleep";
 	pinctrl-0 = <&qspi_clk_pins_a
 		     &qspi_bk1_pins_a
@@ -599,6 +600,41 @@ flash1: flash@1 {
 	};
 };
 
+&qspi_clk_pins_a {
+	bootph-pre-ram;
+	pins {
+		bootph-pre-ram;
+	};
+};
+
+&qspi_bk1_pins_a {
+	bootph-pre-ram;
+	pins {
+		bootph-pre-ram;
+	};
+};
+
+&qspi_cs1_pins_a {
+	bootph-pre-ram;
+	pins {
+		bootph-pre-ram;
+	};
+};
+
+&qspi_bk2_pins_a {
+	bootph-pre-ram;
+	pins {
+		bootph-pre-ram;
+	};
+};
+
+&qspi_cs2_pins_a {
+	bootph-pre-ram;
+	pins {
+		bootph-pre-ram;
+	};
+};
+
 &sai2 {
 	clocks = <&rcc SAI2>, <&rcc PLL3_Q>, <&rcc PLL3_R>;
 	pinctrl-names = "default", "sleep";
diff --git a/arch/arm/boot/dts/st/stm32mp157d-ed1.dts b/arch/arm/boot/dts/st/stm32mp157d-ed1.dts
index d823250956a2..aa333f07d8ba 100644
--- a/arch/arm/boot/dts/st/stm32mp157d-ed1.dts
+++ b/arch/arm/boot/dts/st/stm32mp157d-ed1.dts
@@ -419,6 +419,7 @@ timer@5 {
 };
 
 &uart4 {
+	bootph-all;
 	pinctrl-names = "default", "sleep", "idle";
 	pinctrl-0 = <&uart4_pins_a>;
 	pinctrl-1 = <&uart4_sleep_pins_a>;
@@ -428,6 +429,16 @@ &uart4 {
 	status = "okay";
 };
 
+&uart4_pins_a {
+	bootph-all;
+	pins1 {
+		bootph-all;
+	};
+	pins2 {
+		bootph-all;
+	};
+};
+
 &usbotg_hs {
 	vbus-supply = <&vbus_otg>;
 };
diff --git a/arch/arm/boot/dts/st/stm32mp157d-ev1.dts b/arch/arm/boot/dts/st/stm32mp157d-ev1.dts
index f584420b2007..5c0c7d6d0c0d 100644
--- a/arch/arm/boot/dts/st/stm32mp157d-ev1.dts
+++ b/arch/arm/boot/dts/st/stm32mp157d-ev1.dts
@@ -564,6 +564,7 @@ &m_can1 {
 };
 
 &qspi {
+	bootph-pre-ram;
 	pinctrl-names = "default", "sleep";
 	pinctrl-0 = <&qspi_clk_pins_a
 		     &qspi_bk1_pins_a
@@ -599,6 +600,41 @@ flash1: flash@1 {
 	};
 };
 
+&qspi_clk_pins_a {
+	bootph-pre-ram;
+	pins {
+		bootph-pre-ram;
+	};
+};
+
+&qspi_bk1_pins_a {
+	bootph-pre-ram;
+	pins {
+		bootph-pre-ram;
+	};
+};
+
+&qspi_cs1_pins_a {
+	bootph-pre-ram;
+	pins {
+		bootph-pre-ram;
+	};
+};
+
+&qspi_bk2_pins_a {
+	bootph-pre-ram;
+	pins {
+		bootph-pre-ram;
+	};
+};
+
+&qspi_cs2_pins_a {
+	bootph-pre-ram;
+	pins {
+		bootph-pre-ram;
+	};
+};
+
 &sai2 {
 	clocks = <&rcc SAI2>, <&rcc PLL3_Q>, <&rcc PLL3_R>;
 	pinctrl-names = "default", "sleep";
diff --git a/arch/arm/boot/dts/st/stm32mp157f-ed1.dts b/arch/arm/boot/dts/st/stm32mp157f-ed1.dts
index ef97d301de5b..0f49b89d29e0 100644
--- a/arch/arm/boot/dts/st/stm32mp157f-ed1.dts
+++ b/arch/arm/boot/dts/st/stm32mp157f-ed1.dts
@@ -423,6 +423,7 @@ timer@5 {
 };
 
 &uart4 {
+	bootph-all;
 	pinctrl-names = "default", "sleep", "idle";
 	pinctrl-0 = <&uart4_pins_a>;
 	pinctrl-1 = <&uart4_sleep_pins_a>;
@@ -432,6 +433,16 @@ &uart4 {
 	status = "okay";
 };
 
+&uart4_pins_a {
+	bootph-all;
+	pins1 {
+		bootph-all;
+	};
+	pins2 {
+		bootph-all;
+	};
+};
+
 &usbotg_hs {
 	vbus-supply = <&vbus_otg>;
 };
diff --git a/arch/arm/boot/dts/st/stm32mp157f-ev1.dts b/arch/arm/boot/dts/st/stm32mp157f-ev1.dts
index 41aa9b048f9a..42b05d502387 100644
--- a/arch/arm/boot/dts/st/stm32mp157f-ev1.dts
+++ b/arch/arm/boot/dts/st/stm32mp157f-ev1.dts
@@ -565,6 +565,7 @@ &m_can1 {
 };
 
 &qspi {
+	bootph-pre-ram;
 	pinctrl-names = "default", "sleep";
 	pinctrl-0 = <&qspi_clk_pins_a
 		     &qspi_bk1_pins_a
@@ -600,6 +601,41 @@ flash1: flash@1 {
 	};
 };
 
+&qspi_clk_pins_a {
+	bootph-pre-ram;
+	pins {
+		bootph-pre-ram;
+	};
+};
+
+&qspi_bk1_pins_a {
+	bootph-pre-ram;
+	pins {
+		bootph-pre-ram;
+	};
+};
+
+&qspi_cs1_pins_a {
+	bootph-pre-ram;
+	pins {
+		bootph-pre-ram;
+	};
+};
+
+&qspi_bk2_pins_a {
+	bootph-pre-ram;
+	pins {
+		bootph-pre-ram;
+	};
+};
+
+&qspi_cs2_pins_a {
+	bootph-pre-ram;
+	pins {
+		bootph-pre-ram;
+	};
+};
+
 &sai2 {
 	clocks = <&rcc SAI2>, <&rcc PLL3_Q>, <&rcc PLL3_R>;
 	pinctrl-names = "default", "sleep";
-- 
2.53.0

